ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java

```diff
@@ -794,13 +798,22 @@ private void optimizeTasksCommitted(BasicOptimizeCommit optimizeCommit,
       tasksLock.lock();
       List<OptimizeTaskItem> removedList = new ArrayList<>();
       try {
+        long endTime = System.currentTimeMillis();
         tasks.values().stream().flatMap(Collection::stream).map(OptimizeTaskItem::getTaskId)
             .forEach(optimizeTaskId -> {
               OptimizeTaskItem removed = optimizeTasks.remove(optimizeTaskId);
               if (removed != null) {
                 removedList.add(removed);
                 optimizeTasksMapper.deleteOptimizeTask(optimizeTaskId.getTraceId());
                 internalTableFilesMapper.deleteOptimizeTaskFile(optimizeTaskId);
+
+                // set end time and cost time in optimize_task_history
```

Comment:
Why should we set the cost time after committing?

Reply:
It should be in the optimizeTasksClear method.